systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -540,19 +541,26 @@
     
     % v gradient: dJdot_times_vi/dv
     [dtwistdv, v_indices] = geometricJacobian(model, kinsol, world, i, world);
-    %     dtwistdv = zeros(6, nv);
-    %     dtwistdv(:, v_indices) = J;
     nv_branch = length(v_indices);
     body_branch_velocity_num = nv_branch - length(body.velocity_num) + 1 : nv_branch;
     djoint_twistdv = zeros(6, nv_branch);
+    if isa(kinsol.J{i}, 'msspoly')
```

Comment:
If this will work, sure.  I am not 100% sure I understand it but I will
test it when I get a chance.  I was not aware that isa has huge performance
issues - I thought that was the recommended way to do it, but if that's
slow we should definitely rid of it.

-Andy S.


On Thu, Jul 30, 2015 at 11:10 PM, Twan Koolen <notifications@github.com>
wrote:

> In systems/plants/@RigidBodyManipulator/doKinematics.m
> <https://github.com/RobotLocomotion/drake/pull/1212#discussion_r35943717>:
>
> >      nv_branch = length(v_indices);
> >      body_branch_velocity_num = nv_branch - length(body.velocity_num) + 1 : nv_branch;
> >      djoint_twistdv = zeros(6, nv_branch);
> > +    if isa(kinsol.J{i}, 'msspoly')
>
> can we change these into something like
>
> djoint_twistdv = zeros(6, nv_branch) * J{i}(1)
>
> It's ugly, but using isa is very slow.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RobotLocomotion/drake/pull/1212/files#r35943717>.
>